manager/machine.go

```diff
@@ -210,12 +210,17 @@ func getTopology(sysFs sysfs.SysFs, cpuinfo string) ([]info.Node, int, error) {
 }
 
 func getMachineID() string {
-	id, err := ioutil.ReadFile(*machineIdFilePath)
-	if err != nil {
-		glog.Error("Couldn't collect machine-id: ", err)
+	if len(*machineIdFilePath) == 0 {
 		return ""
 	}
-	return strings.TrimSpace(string(id))
+	for _, file := range strings.Split(*machineIdFilePath, ",") {
+		id, err := ioutil.ReadFile(file)
+		if err == nil {
+			return strings.TrimSpace(string(id))
+		}
+	}
+	glog.Infof("Couldn't collect machine-id from any of the files in %q", *machineIdFilePath)
```

Comment:
I am not sure if we have machine-id on all distro. It might be expected to fail on some hosts.